packages/vats/src/proposals/vtransfer-echoer.js

```diff
@@ -0,0 +1,44 @@
+// @ts-check
+import { makeExo } from '@agoric/store';
+import { E } from '@endo/far';
+
+/**
+ * @param {BootstrapPowers & {
+ *   consume: {
+ *     transferMiddleware: import('../vat-transfer.js').TransferMiddleware;
+ *   };
+ * }} powers
+ * @param {object} options
+ * @param {{ target: string }} options.options
+ */
+export const echoVtransfer = async (
+  { consume: { transferMiddleware } },
+  { options: { target } },
+) => {
+  console.warn(`=== vtransfer echoer targeting ${target}`);
+
+  // a tap that simply returns what it received
+  const tap = makeExo('echoer', undefined, {
+    // ack value must be stringlike
+    upcall: async param => JSON.stringify(param),
+  });
+
+  // TODO put something in promise space to unregister this one and register again
+  // or maybe put all imperative stuff into promise space
+  await E(transferMiddleware).registerActiveTap(target, tap);
```

Comment:
Is this TODO expected to be part of this PR?

Reply:
Reiterating https://github.com/Agoric/agoric-sdk/pull/8624/files#r1602758933 (cc @turadg):
> Is this TODO expected to be part of this PR?

Reply:
Sorry I had missed the mentions.

It's not hard to add things to promise space later so if we don't have a use case requiring that yet I think we should defer. I'd take out the TODO though. If you think it should be done at some point, file an issue and we can figure out what will motivate adding it.